hummingbot/connector/derivative/hyperliquid_perpetual/hyperliquid_perpetual_derivative.py

```diff
@@ -0,0 +1,789 @@
+
+    @property
+    def domain(self) -> str:
+        return self._domain
+
+    @property
+    def client_order_id_max_length(self) -> int:
+        return CONSTANTS.MAX_ORDER_ID_LEN
+
+    @property
+    def client_order_id_prefix(self) -> str:
+        return CONSTANTS.BROKER_ID
+
+    @property
+    def trading_rules_request_path(self) -> str:
+        return CONSTANTS.EXCHANGE_INFO_URL
+
+    @property
+    def trading_pairs_request_path(self) -> str:
+        return CONSTANTS.EXCHANGE_INFO_URL
+
+    @property
+    def check_network_request_path(self) -> str:
+        return CONSTANTS.PING_URL
+
+    @property
+    def trading_pairs(self):
+        return self._trading_pairs
+
+    @property
+    def is_cancel_request_in_exchange_synchronous(self) -> bool:
+        return True
+
+    @property
+    def is_trading_required(self) -> bool:
+        return self._trading_required
+
+    @property
+    def funding_fee_poll_interval(self) -> int:
+        return 120
+
+    async def _make_network_check_request(self):
+        await self._api_post(path_url=self.check_network_request_path, data={"type": CONSTANTS.META_INFO})
+
+    def supported_order_types(self) -> List[OrderType]:
+        """
+        :return a list of OrderType supported by this connector
+        """
+        return [OrderType.LIMIT, OrderType.LIMIT_MAKER, OrderType.MARKET]
+
+    def supported_position_modes(self):
+        """
+        This method needs to be overridden to provide the accurate information depending on the exchange.
+        """
+        return [PositionMode.ONEWAY]
+
+    def get_buy_collateral_token(self, trading_pair: str) -> str:
+        trading_rule: TradingRule = self._trading_rules[trading_pair]
+        return trading_rule.buy_order_collateral_token
+
+    def get_sell_collateral_token(self, trading_pair: str) -> str:
+        trading_rule: TradingRule = self._trading_rules[trading_pair]
+        return trading_rule.sell_order_collateral_token
+
+    def _is_request_exception_related_to_time_synchronizer(self, request_exception: Exception):
+        return False
+
+    def _create_web_assistants_factory(self) -> WebAssistantsFactory:
+        return web_utils.build_api_factory(
+            throttler=self._throttler,
+            auth=self._auth)
+
+    async def _make_trading_rules_request(self) -> Any:
+        exchange_info = await self._api_post(path_url=self.trading_rules_request_path,
+                                             data={"type": CONSTANTS.ASSET_CONTEXT_TYPE})
+        return exchange_info
+
+    async def _make_trading_pairs_request(self) -> Any:
+        exchange_info = await self._api_post(path_url=self.trading_pairs_request_path,
+                                             data={"type": CONSTANTS.ASSET_CONTEXT_TYPE})
+        return exchange_info
+
+    def _is_order_not_found_during_status_update_error(self, status_update_exception: Exception) -> bool:
+        # TODO: implement this method correctly for the connector
```

Comment:
we need to implement these methods to prevent the bot getting stucked asking for an order that does not exist

Reply:
added